### DISCUSSION TITLE
Migration from tac_plus version F4.0.4.27a

### DISCUSSION THREAD

**Discussion by enoch85** [MarcJHuber/event-driven-servers#83] [General]
Hi,

We are running tac_plus version F4.0.4.27a on a Debian server, and are in the process of moving away from that server due to legacy OS etc...

I'm just wondering if there's a good "migration path" to tac_plus-ng? From what I've read the requirements are different (less), and the config files looks a bit different..?

I'm new to TACACS+ so please bear with me. :)

**Comment by MarcJHuber** [comment]
Hi,

if I remember correctly, tac_plus F4... configuration is pretty close to the original Cisco tac_plus syntax. For migration, I'd start with one of the sample configurations provided and align the obvious "host" and "user" configuration parameters (keys, IP addresses, passwords) and assign users to groups.

Other things are pretty much different. E.g., tac_plus-ng got away with service definitions by using profiles and a basic scripting language to model the authorization flow. I'd have a look at the [sample configuration](https://www.pro-bono-publico.de/projects/unpacked/tac_plus-ng/sample/tac_plus-ng.cfg), and [the documentation](https://www.pro-bono-publico.de/projects/tac_plus-ng.html#AEN160) might give pretty good starting point, too.

All in all, the complexity of migration pretty much depends on the complexity your existing configuration. 

Cheers,

Marc

**Comment by enoch85** [comment]
Thanks! Will have a look at this.

Keeping this open if further questions arise.

**Comment by enoch85** [comment]
@MarcJHuber We finally came around to migration, and guess what - we succeeded. Great work by you!

Now, when we want to upgrade, how do we do that? Just pull it again and compile?

Our current setup looks like this:


``````
root@hostac01:/usr/local/etc# ls -l
total 16
drwxr-xr-x 3 root root 4096 Oct 15 11:16 mavis
-rwxr-xr-x 1 root root 5067 Oct 15 11:16 tac_plus-ng.SAMPLE
-rw-r--r-- 1 root root 3153 Oct 30 08:20 tac_plus-ng.cfg
``````

**Comment by MarcJHuber** [comment]
Hi,

yes, pulling and rebuilding will work. I'd run "./configure ... (whatever options you initial used) && make clean all && sudo make install"

Cheers,

Marc
